hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -232,7 +232,33 @@ extends Importer:
       term(rhs)
     case tree @ App(lhs, rhs) =>
       val sym = FlowSymbol("‹app-res›", nextUid)
-      Term.App(term(lhs), term(rhs))(tree, sym)
+      val lt = term(lhs)
+      val rt = term(rhs)
+
+      // Check module parameters
+      val args = rt match
+        case Term.Tup(fields) => S(fields)
+        case _ => N
+      val argsModFlags = args
+        .map(_.map(_.flags.mod))
+      val paramsModFlags = lt.symbol match
+        case S(sym: BlockMemberSymbol) => sym.defn match
+          case S(defn: TermDefinition) => defn.params.lift(0)
+            .map(_.params.map(_.flags.mod))
+          case _ => argsModFlags.map(_.map(_ => false))
+        case _ => argsModFlags.map(_.map(_ => false))
```

Comment:
This logic is very strange/fishy. In general, I also don't like all that mapping just to consume it in the `for` below. Better zip the sources directly and destructure the corresponding things in the `for`.